xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/Timers.h

```diff
@@ -0,0 +1,816 @@
+  unsigned int GetId() const { return m_cStructure->iId; }
+
+  /// @brief **required**\n
+  /// Defines the attributes for this type (@ref cpp_kodi_addon_pvr_Defs_Timer_PVR_TIMER_TYPE "PVR_TIMER_TYPE_*" constants).
+  ///
+  /// To defines the attributes for a type. These values are bit fields that can be
+  /// used together.
+  ///
+  ///--------------------------------------------------------------------------
+  ///
+  /// **Example:**
+  /// ~~~~~~~~~~~~~{.cpp}
+  /// kodi::addon::PVRTimerType tag;
+  /// tag.SetAttributes(PVR_TIMER_TYPE_IS_MANUAL | PVR_TIMER_TYPE_IS_REPEATING);
+  /// ~~~~~~~~~~~~~
+  ///
+  void SetAttributes(unsigned int attributes) { m_cStructure->iAttributes = attributes; }
+
+  /// @brief To get with @ref SetAttributes changed values.
+  unsigned int GetAttributes() const { return m_cStructure->iAttributes; }
+
+  /// @brief **optional**\n
+  /// A short localized string describing the purpose of the type. (e.g.
+  /// "Any time at this channel if title matches").
+  ///
+  /// If left blank, Kodi will generate a description based on the attributes
+  /// REPEATING and MANUAL. (e.g. "Repeating EPG-based.")
+  void SetDescription(const std::string& description)
+  {
+    strncpy(m_cStructure->strDescription, description.c_str(),
+            sizeof(m_cStructure->strDescription) - 1);
+  }
+
+  /// @brief To get with @ref SetDescription changed values.
+  std::string GetDescription() const { return m_cStructure->strDescription; }
+
+  /// @brief **optional**\n
+  /// Priority value definitions.
+  ///
+  /// Array containing the possible values for @ref PVRTimer::SetPriority().
+  ///
+  /// @param[in] priorities List of priority values
+  /// @param[in] prioritiesDefault [opt] The default value in list, can also be
+  ///                              set by @ref SetPrioritiesDefault()
+  void SetPriorities(const std::vector<std::pair<int, std::string>>& priorities,
+                     int prioritiesDefault = -1)
+  {
+    m_cStructure->iPrioritiesSize = priorities.size();
+    for (unsigned int i = 0;
+         i < m_cStructure->iPrioritiesSize && i < sizeof(m_cStructure->priorities); ++i)
+    {
+      m_cStructure->priorities[i].iValue = priorities[i].first;
+      strncpy(m_cStructure->priorities[i].strDescription, priorities[i].second.c_str(),
+              sizeof(m_cStructure->priorities[i].strDescription) - 1);
+    }
+    if (prioritiesDefault != -1)
+      m_cStructure->iPrioritiesDefault = prioritiesDefault;
+  }
+
+  /// @brief To get with @ref SetPriorities changed values.
+  std::vector<std::pair<int, std::string>> GetPriorities() const
+  {
+    std::vector<std::pair<int, std::string>> ret;
+    for (unsigned int i = 0; i < m_cStructure->iPrioritiesSize; ++i)
+    {
+      int value = m_cStructure->priorities[i].iValue;
+      std::string description = m_cStructure->priorities[i].strDescription;
+      ret.push_back(std::pair<int, std::string>(value, description));
```

Comment:
`ret.emplace_back(` - always prefer emplace_back over push_back. Applies to all changes done by this PR